### PR TITLE
upgrade_config checks for nested changes

### PR DESCRIPTION
**Pull request by dbxnr**
Fix for #714 - the upgrade_config algorithm recursively replaces missing key/value pairs with default values.

### Checklist
- [x] The code change is tested and works locally.
- [x] Tests pass. Your PR cannot be merged unless tests pass
- [x] There is no commented out code in this PR.
- [x] Have you followed the guidelines in our Contributing document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?

### CODE REVIEW

File: jrnl/install.py [hunk 1]
```diff
@@ -61,19 +62,16 @@ def module_exists(module_name):
...
-        save_config(config)
-        print(
-            f"[Configuration updated to newest version at {CONFIG_FILE_PATH}]",
-            file=sys.stderr,
-        )
+    for k, v in new_config.items():
+        new_value = user_config.get(k)
+        if isinstance(v, Mapping) and isinstance(new_value, Mapping):
+            user_config[k] = v
+            upgrade_config(new_value, v)
```
Comment (wren) [inline jrnl/install.py]: Good use of recursion.

File: jrnl/install.py [hunk 2]
```diff
@@ -113,7 +111,9 @@ def load_or_install_jrnl():
             print("Exiting.", file=sys.stderr)
             sys.exit(1)
 
-        upgrade_config(config)
+        if config.get("version") and config.get("version") < __version__:
```
Comment (wren) [inline jrnl/install.py]: Why the change in behavior here?

Comment (dbxnr) [reply jrnl/install.py]: Optimisation, without this check it would try to upgrade the config file on every invocation

**Comment by dbxnr** [comment]
I'm really not sure I can test it with behave while default_config is hard-coded into install.py unless I'm missing something?

There are quite a few environmental variables that probably shouldn't be in there, moving them to jrnl.conf would make it testable... What are your thoughts?
